File: tests/fixtures/code-query/sample.py
def greet(name: str) -> str:
    return "hi " + name


class User:
    def __init__(self, name: str) -> None:
        self.name = name

    def greet(self) -> str:
        return greet(self.name)


u = User("world")
print(u.greet())

File: tests/fixtures/code-query/sample.rs
pub struct User {
    pub name: String,
}

pub trait Greet {
    fn greet(&self) -> String;
}

pub fn hello(name: &str) -> String {
    format!("hi {}", name)
}

impl Greet for User {
    fn greet(&self) -> String {
        hello(&self.name)
    }
}

pub fn main() {
    let u = User { name: String::from("world") };
    println!("{}", u.greet());
}
